src/slice.cr

```diff
@@ -244,8 +244,6 @@ struct Slice(T)
   # ```
   # a = Slice["a", "b", "c", "d", "e"]
   # a[1..3] # => Slice["b", "c", "d"]
-  # # range.end > array.size
-  # a[3..7] # => Slice["d", "e"]
```

Comment:
This is not correct because `a[3..7]` raises `IndexError`.
And since the usage are written right after here, I simply removed the code here.